src/core_plugins/kibana/public/dashboard/top_nav/save_modal.js

```diff
@@ -0,0 +1,252 @@
+          </p>
+        </EuiCallOut>
+        <EuiSpacer />
+      </Fragment>
+    );
+  }
+
+  renderCopyOnSave = () => {
+    if (!this.props.showCopyOnSave) {
+      return;
+    }
+
+    let helpText;
+    if (this.props.title !== this.state.title) {
+      helpText = `In previous versions of Kibana, changing the name of a dashboard would make a copy with the new name.
+      Use the 'Save as a new dashboard' checkbox to do this now.`;
+    }
+
+    return (
+      <EuiFormRow
+        label="Save as a new dashboard"
+        helpText={helpText}
+      >
+        <EuiSwitch
+          data-test-subj="saveAsNewCheckbox"
+          checked={this.state.copyOnSave}
+          onChange={this.onCopyOnSaveChange}
+        />
+      </EuiFormRow>
+    );
+  }
+
+  render() {
+    return (
+      <EuiOverlayMask>
+        <EuiModal
+          data-test-subj="dashboardSaveModal"
+          className="dashboardModal"
+          onClose={this.props.onClose}
+        >
+          <EuiModalHeader>
+            <EuiModalHeaderTitle>
+              Save Dashboard
+            </EuiModalHeaderTitle>
+          </EuiModalHeader>
+
+          <EuiModalBody>
+
+            {this.renderDuplicateTitleCallout()}
+
+            <EuiForm>
+
+              {this.renderCopyOnSave()}
+
+              <EuiFormRow
+                label="Title"
+              >
+                <EuiFieldText
+                  autoFocus
+                  data-test-subj="dashboardTitle"
+                  value={this.state.title}
+                  onChange={this.onTitleChange}
+                  isInvalid={this.state.hasTitleDuplicate}
+                />
+              </EuiFormRow>
+
+              <EuiFormRow
+                label="Description"
+              >
+                <EuiTextArea
+                  data-test-subj="dashboardDescription"
+                  value={this.state.description}
+                  onChange={this.onDescriptionChange}
+                  compressed
+                />
+              </EuiFormRow>
+
+              <EuiFormRow
+                label="Store time with dashboard"
+                helpText="This changes the time filter to the currently selected time each time this dashboard is loaded."
+              >
+                <EuiSwitch
+                  data-test-subj="storeTimeWithDashboard"
+                  checked={this.state.timeRestore}
+                  onChange={this.onTimeRestoreChange}
+                />
+              </EuiFormRow>
+
+            </EuiForm>
+
+          </EuiModalBody>
+
+          <EuiModalFooter>
+            <EuiButton
+              data-test-subj="saveCancelButton"
+              onClick={this.props.onClose}
+            >
+              Cancel
+            </EuiButton>
+
+            <EuiButton
+              fill
+              data-test-subj="confirmSaveDashboardButton"
+              onClick={this.saveDashboard}
+              isLoading={this.state.isLoading}
```

Comment:
Don't know if this is working for me. I'm able to click it a bunch of times in a row, and if `save as new` is checked then it can create a ton of dashboards.

<img width="704" alt="screen shot 2018-05-30 at 12 22 42 pm" src="https://user-images.githubusercontent.com/16563603/40733726-60413996-6404-11e8-9f29-1cd3d55187d0.png">
![Uploading Screen Shot 2018-05-30 at 12.22.49 PM.png…]()